src/openrct2-ui/windows/ScenarioSelect.cpp

```diff
@@ -809,31 +738,46 @@ static void InitialiseListItems(WindowBase* w)
     }
 }
 
-static bool IsScenarioVisible(WindowBase* w, const ScenarioIndexEntry* scenario)
+void DrawCategoryHeading(WindowBase* w, DrawPixelInfo* dpi, int32_t left, int32_t right, int32_t y, StringId stringId)
 {
-    if (gConfigGeneral.ScenarioSelectMode == SCENARIO_SELECT_MODE_ORIGIN || _titleEditor)
-    {
-        if (static_cast<uint8_t>(scenario->SourceGame) != w->selected_tab)
-        {
-            return false;
-        }
-    }
-    else
-    {
-        int32_t category = scenario->Category;
-        if (category > SCENARIO_CATEGORY_OTHER)
-        {
-            category = SCENARIO_CATEGORY_OTHER;
-        }
-        if (category != w->selected_tab)
-        {
-            return false;
-        }
-    }
-    return true;
+    colour_t baseColour = w->colours[1];
+    colour_t lightColour = ColourMapA[baseColour].lighter;
+    colour_t darkColour = ColourMapA[baseColour].mid_dark;
+
+    // Draw string
+    int32_t centreX = (left + right) / 2;
+    DrawTextBasic(*dpi, { centreX, y }, stringId, {}, { baseColour, TextAlignment::CENTRE });
+
+    // Get string dimensions
+    utf8 buffer[CommonTextBufferSize];
+    auto bufferPtr = buffer;
+    OpenRCT2::FormatStringLegacy(bufferPtr, sizeof(buffer), stringId, nullptr);
+    int32_t categoryStringHalfWidth = (GfxGetStringWidth(bufferPtr, FontStyle::Medium) / 2) + 4;
+    int32_t strLeft = centreX - categoryStringHalfWidth;
+    int32_t strRight = centreX + categoryStringHalfWidth;
+
+    // Draw light horizontal rule
+    int32_t lineY = y + 4;
+    auto lightLineLeftTop1 = ScreenCoordsXY{ left, lineY };
+    auto lightLineRightBottom1 = ScreenCoordsXY{ strLeft, lineY };
+    GfxDrawLine(dpi, { lightLineLeftTop1, lightLineRightBottom1 }, lightColour);
+
+    auto lightLineLeftTop2 = ScreenCoordsXY{ strRight, lineY };
+    auto lightLineRightBottom2 = ScreenCoordsXY{ right, lineY };
+    GfxDrawLine(dpi, { lightLineLeftTop2, lightLineRightBottom2 }, lightColour);
+
+    // Draw dark horizontal rule
+    lineY++;
+    auto darkLineLeftTop1 = ScreenCoordsXY{ left, lineY };
+    auto darkLineRightBottom1 = ScreenCoordsXY{ strLeft, lineY };
+    GfxDrawLine(dpi, { darkLineLeftTop1, darkLineRightBottom1 }, darkColour);
+
+    auto darkLineLeftTop2 = ScreenCoordsXY{ strRight, lineY };
+    auto darkLineRightBottom2 = ScreenCoordsXY{ right, lineY };
+    GfxDrawLine(dpi, { darkLineLeftTop2, darkLineRightBottom2 }, darkColour);
 }
 
-static bool IsLockingEnabled(WindowBase* w)
+bool IsLockingEnabled(WindowBase* w)
```

Comment:
Why has this been made not `static`? There are a bunch of methods that used to be `static`, but have been made member functions, which I don't get why, since they still receive the parameters needed. I'd either keep them as members, but trim down parameters like `WindowBase` or keep them as `static`s out of the class itself.

Reply:
converted these back to static